Prefix CMake options with RAJA_ (#1119)
LLNL/RAJA: docs/sphinx/user_guide/config_options.rst

RAJA uses `BLT <https://github.com/LLNL/blt>`_, a CMake-based build system.
In :ref:`getting_started-label`, we described how to run CMake to configure
RAJA with its default option settings. In this section, we describe RAJA
configuration options that are most useful for users to know about and
their defaults.

=============================
RAJA Option Types
=============================

All Cmake options used in RAJA contains the prefix ``RAJA_`` to give users 
the flexibility to enable/disable individual options for RAJA, specifically.
RAJA contains two types of options, those that only exist in RAJA

> and those that are similar tostandard CMake options or options 

Reviewer: ```suggestion
and those that are similar to the standard CMake options or options 
```

Reviewer 2: Thanks. Fixed.

      =========================   ======================
      Variable                    Default
      =========================   ======================

>       RAJA_ENABLE_WARNINGS_AS_ERRORS   Off

Reviewer 2: I may be wrong, but think these changes are going to break the sphinx docs formatting. As I understand it, the '=' row delimeters need to extend at least as far as the test in each row. Adding more '=' so the columns align should fix it. Have you looked at the regenerated docs with these changes?

Reviewer 2: Indeed, running text into the column margins breaks the sphinx build. I pushed changes to fix.

Reviewer: Thanks Rich!

                                   also be on!)
      RAJA_ENABLE_TBB              Off
      RAJA_ENABLE_SYCL             Off
      ==========================   ============================================

Other programming model specific compilation options are also available:

      ======================================   =================================
      Variable                                 Default
      ======================================   =================================
      (RAJA_)ENABLE_CLANG_CUDA                 Off (if on, RAJA_ENABLE_CUDA 
                                               must be on too!)
      RAJA_ENABLE_EXTERNAL_CUB                 Off
      RAJA_ENABLE_NV_TOOLS_EXT                 Off
      CUDA_ARCH                                sm_35 (based on hardware support)
      RAJA_ENABLE_EXTERNAL_ROCPRIM             Off
      RAJA_ENABLE_ROCTX                        Off
      RAJA_ENABLE_HIP_INDIRECT_FUNCTION_CALL   Off (enables device function 
                                               pointers in Hip back-end)
      ======================================   =================================

> Turning the ``(RAJA)_ENABLE_CLANG_CUDA`` variable on will build CUDA 

Reviewer: ```suggestion
Turning the ``(RAJA_)ENABLE_CLANG_CUDA`` variable on will build CUDA 
```

Reviewer: Found one more!

RAJA features. Most variables get translated to 
compiler directives and definitions in the RAJA ``config.hpp`` file that is 
generated when CMake runs. The ``config.hpp`` header file is included in other 
RAJA headers so all options propagate consistently through the 
build process for all of the code. 

.. note:: The following discussion indicates which options exist in RAJA 
          only and those which are dependent options. **Dependent options 
          appear in the discussion with parentheses around the ``RAJA_`` 

>           prefix to indicate that the option name without the prefix can be 

Reviewer 2: Good catch. Fixed. Docs should be served up in a few minutes.

          used and is often preferred.**

The following tables describe which variables set RAJA options and 
and their default settings:

* **Examples, tests, warnings, etc.**

     CMake variables can be used to control whether RAJA tests, examples, 
     tutorial exercises, etc. are built when RAJA is compiled.

      =========================  =========================================
      Variable                   Default
      =========================  =========================================

>       (RAJA)_ENABLE_TESTS        On 

Reviewer: Shouldn't the parentheses be around "RAJA_" not just "RAJA"?

Reviewer 2: Grrff... I'm fixing other issues in the docs and am not being careful. Fixes coming.

* **Examples, tests, warnings, etc.**

     CMake variables can be used to control whether RAJA tests, examples, 
     tutorial exercises, etc. are built when RAJA is compiled.

.. note:: In the following tables, an option name followed by an asterisk ('*')
          only exists in RAJA (i.e., no associated CMake or BLT variant). 
          For the others (dependent options), it is recommended to use the
          non-RAJA form (with ``RAJA_`` prefix removed) to enable features.

      ========================   =========================================
      Variable                   Default
      ========================   =========================================

>       RAJA_ENABLE_TESTS          On 

Author: ```suggestion
      (RAJA_)ENABLE_TESTS          On 
```

Do you think using parentheses for the dependent options might be clearer still? The most confusing thing (that we want to avoid) is that for something like CUDA, you _have_ to say ENABLE_CUDA=On (because RAJA_ENABLE_CUDA=On won't change the BLT option)

Reviewer 2: That's a good idea. I will work on that change. Thanks.
